### PR TITLE
Change call_order_update_operation to return order_id #1269

### PR DESCRIPTION
**Pull request by cogutvalera**
PR for #1269 **"Change call_order_update_operation to return order_id"**

### CODE REVIEW

File: libraries/chain/market_evaluator.cpp [hunk 1]
```diff
@@ -237,18 +238,20 @@ void_result call_order_update_evaluator::do_apply(const call_order_update_operat
...
       auto new_collateral = call_obj->collateral + o.delta_collateral.amount;
       auto new_debt = call_obj->debt + o.delta_debt.amount;
+      call_order_id = call_obj->id;
 
       if( new_debt == 0 )
       {
          FC_ASSERT( new_collateral == 0, "Should claim all collateral when closing debt position" );
          d.remove( *call_obj );
-         return void_result();
+         return call_order_id;
```
Comment (jmjatlanta) [inline libraries/chain/market_evaluator.cpp]: Should we return the id of a deleted object?

File: libraries/chain/market_evaluator.cpp [hunk 2]
```diff
@@ -264,13 +267,12 @@ void_result call_order_update_evaluator::do_apply(const call_order_update_operat
                                               _bitasset_data->current_feed.maintenance_collateral_ratio );
          call.target_collateral_ratio = o.extensions.value.target_collateral_ratio;
       });
+      call_order_id = call_obj->id;
```
Comment (pmconrad) [inline libraries/chain/market_evaluator.cpp]: This is superfluous, modify() will not change the id.

File: libraries/chain/market_evaluator.cpp (hunk 1 above)
Comment (pmconrad) [reply libraries/chain/market_evaluator.cpp]: Agree, makes sense to return it.

**Comment by cogutvalera** [comment]
any other changes must be here or we can approve and merge this PR ?

Thanks a lot !

**Comment by cogutvalera** [comment]
Thanks for review ! Thank you very much for your time and efforts !

**Pull request merged**